### PR TITLE
chore: extract test-app into own package

### PR DESCRIPTION
**Pull request by SergeAstapov**
@marcoow this is continuation of the work started in #740

This PR introduces new package in the monorepo `test-app` that holds the dummy test app and tests.

Removed from the `packages/ember-cookies` app related code and from the `packages/test-app` addon related code (however tried to do just minimum changes for dependencies and eslintrc file).

Some files marked as "new" nut they are actually copied from `packages/ember-cookies` to `packages/test-app`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,10 +27,15 @@ jobs:
...
-      - name: Lint
+
+      - name: Lint Addon
         run: yarn lint:js
         working-directory: packages/ember-cookies
 
+      - name: Lint Test App
+        run: yarn lint:js
+        working-directory: packages/test-app
+
```
Comment (marcoow) [inline .github/workflows/ci.yml:38]: I think the linting of the addon and app should run in parallel – currently I'm not sure the lint job for the app would run at all if the one for the addon fails?

**Review by marcoow — COMMENTED** [review]
Nice 🙌

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SergeAstapov) [reply .github/workflows/ci.yml:38]: @marcoow ok, I can do it: would you like to introduce two new lint jobs: for addon and test app *or* only extract addon lint into separate job so test app lint + tests grouped?

Comment (SergeAstapov) [reply .github/workflows/ci.yml:33]: @marcoow this is solved - the linting of the addon and app should run in parallel and is setup via `lint` script in the root `package.json`

**Comment by SergeAstapov** [comment]
@marcoow do you mind to take another look or delegate if there is anyone available?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcoow) [reply .github/workflows/ci.yml:33]: It still seems like we're running at least the linting sequentually? https://github.com/simplabs/ember-cookies/runs/6499932915?check_suite_focus=true

Comment (marcoow) [reply .github/workflows/ci.yml:33]: Yeah, I'd split into separate jobs 👍

Comment (SergeAstapov) [reply .github/workflows/ci.yml:52]: @marcoow did split into `lint-addon` and `lint-test-app` jobs that run in parallel on separate agents. Does this look good to you?

**Review by marcoow — APPROVED** [review]
👍

**Pull request merged**